In [ ]:
### =b

### ชนิดสมมติฐาน
1. สมมติฐานหบัก (null, H<sub>0</sub>) : ค่าพารามีเตอร์ มีค่าเท่ากับค่าใดค่าหนึ่ง (=, <=, >=)
2. รอง (alternative, H<sub>1</sub> or H<sub>3</sub>) : ค่าพารามีเตอร์ มีค่าแตกต่างจากสมมติฐาน (!=, <, >)

#### ขั้นตอนการทดวอบสมมติฐาน

1. ตั้งสมมติฐาน H<sub>0</sub> และ H<sub>1</sub>
2. กำหนดระดับนัยสำคัญ () lamda = 0.05, 0.01, 0.001
3. คำนวณ Z<sub>cal</sub> หรือ t<sub>cal</sub> จากข้อมูลที่มี แล้วหาค่า p-value
4. สรุปผลการทดสอบสมมติฐาน


>$p-value <= α$ : **Deny** H<sub>0</sub>\
 $p-value > α$ : **Accept** H<sub>0</sub>

In [2]:
import numpy as np
from scipy import stats
#from statsmodels.stats.weightstats import ztest

#### Ex 8.1 
![image](../imgs/example8_3.png)

กำหนด μ = คือ นำ้หนักเฉลี่ยของนผลไม้กระป๋อง\

μ<sub>0</sub> = 160
1. ตั้งสมมติฐาน\
    H<sub>0</sub> : μ = μ<sub>0</sub>\
    H<sub>1</sub> : μ != μ<sub>0</sub>
2. ระดัยนัยสำคัญ : 	α = 0.05
3. ไม่มีความแปรปรวน ตัวอย่างขนาดเล็ก n < 30 => ใช้การทดสอบด้วย t
$$ t_{cal} = \frac{\bar{x} - \mu}{\frac{s}{\sqrt{n}}},    v = n - 1 $$

In [7]:
# คำนวณหา tcal
weight = [164, 161, 158, 170, 161, 152, 158, 169, 152, 160]
x_bar = np.mean(weight)
s = np.std(weight, ddof=1)
alpha = 0.05
mu_0 = 160
print('x_bar = ', x_bar)

t_cal = (x_bar - mu_0) / (s / np.sqrt(len(weight)))
print('t_cal = ', t_cal)

x_bar =  160.5
t_cal =  0.260132990857236


In [8]:
v = len(weight) - 1
t_crit_left = stats.t.ppf(alpha / 2, df=v)
t_crit_right = stats.t.isf(alpha / 2 , df=v)

print('t_crit_left = ', t_crit_left)
print('t_crit_right = ', t_crit_right)

t_crit_left =  -2.262157162740992
t_crit_right =  2.262157162740992


In [9]:
p_value = 2 * stats.t.sf(np.abs(t_cal), df=v)
p_value

0.8006146644674419

**ดังนั้น** p-value = 0.8006 > 0.05 (alpha) => ไม่ปฏิเสธสมมติฐาน H<sub>0</sub> ที่ μ = μ<sub>0</sub>

#### Ex 8.2

### ค่าเฉลี่ยของประชากร 2 กลุ่มอิสระต่อกัน

มี $\mu_1$ และ $\mu_2$ คือค่าเฉลี่ยของกลุ่ม 1 และ 2 ตามลำดับ และ $\mu_0 = 0$ เป็นพารามีเตอร์ที่ต้องการทดสอบ 

เมื่อ $\mu_0 = 0$
1. Two-tailed test:\ 
$H_0: \mu_1 - \mu_2 = 0$  or  $H_0: \mu_1 = \mu_2$\
$H_1: \mu_1 - \mu_2 \neq 0$ or $H_1: \mu_1 \neq \mu_2$

2. Left-tailed test:\
$H_0: \mu_1 - \mu_2 \geq 0$  or  $H_0: \mu_1 \geq \mu_2$\
$H_1: \mu_1 - \mu_2 < 0$ or $H_1: \mu_1 < \mu_2$

3. Right-tailed test:\
$H_0: \mu_1 - \mu_2 \leq 0$  or  $H_0: \mu_1 \leq \mu_2$\
$H_1: \mu_1 - \mu_2 > 0$ or $H_1: \mu_1 > \mu_2$

#### สถิตติกส์ที่ใช้ในการทดสอบสมมติฐาน
1. ทราบ $\sigma_1$ และ $\sigma_2$ ของทั้ง 2 กลุ่ม\
    > $ Z = \frac{(\bar{x}_1 - \bar{x}_2) - (\mu_1 - \mu_2)}{\sqrt{\frac{\sigma_1^2}{n_1} + \frac{\sigma_2^2}{n_2}}} $

2. ไม่ทราบ $\sigma_1$ และ $\sigma_2$ ของทั้ง 2 กลุ่ม และตัวอย่าง\  $n_1, n_2 \geq 30$
    > $ z = \frac{(\bar{x}_1 - \bar{x}_2) - (\mu_1 - \mu_2)}{\sqrt{\frac{s_1^2}{n_1} + \frac{s_2^2}{n_2}}} $

3. ไม่ทราบ $\sigma_1$ และ $\sigma_2$ ของทั้ง 2 กลุ่ม และตัวอย่าง\  $n_1, n_2 < 30$ แต่ทราบว่า $\sigma_1 = \sigma_2$
    > $ t = \frac{(\bar{x}_1 - \bar{x}_2) - (\mu_1 - \mu_2)}{s^2_p \sqrt{\frac{1}{n_1} + \frac{1}{n_2}}},   $ $d.f. = n_1 + n_2 - 2$\
    > $ s^2_p = \sqrt{\frac{(n_1 - 1)s_1^2 + (n_2 - 1)s_2^2}{n_1 + n_2 - 2}} $

4. ไม่ทราบความแปรปรวนของประชากร $\sigma^2_1$ และ $\sigma^2_2$ ตัวอย่างมีขนาดเล็ก $n_1, n_2 < 30$ และ $\sigma_1 \neq \sigma_2$
    > $ t = \frac{(\bar{x}_1 - \bar{x}_2) - (\mu_1 - \mu_2)}{\sqrt{\frac{s_1^2}{n_1} + \frac{s_2^2}{n_2}}},   $ $d.f. = \frac{(\frac{s_1^2}{n_1} + \frac{s_2^2}{n_2})^2}{\frac{1}{n_1 - 1}(\frac{s_1^2}{n_1})^2 + \frac{1}{n_2 - 1}(\frac{s_2^2}{n_2})^2}$

### สัดส่วนประชากร 1 กลุ่ม

$ 
    Z = p\hat{p} \sqrt{\frac{1}{n_1} + \frac{1}{n_2}}
$